parlai/core/utils.py

```diff
@@ -181,6 +181,28 @@ def load_opt_file(optfile):
     return Opt(opt)
 
 
+class Message(dict):
+    """
+    Class for observations and actions.
+
+    Functions like a dict, but breaks when writing to fields that already exist.
+    """
+
+    def __setitem__(self, key, val):
+        if key in self:
+            raise RuntimeError(
+                'Message already contains key `{}`. If this was intentional, '
+                'please use the function `set_key(key, value)`.'.format(key)
+            )
+        super().__setitem__(key, val)
+
+    def force_set(self, key, val):
+        super().__setitem__(key, val)
+
+    def copy(self):
+        return Message(self)
```

Comment:
I think we're also going to need to support deepcopy.

Reply:
it should automatically support deepcopy, there are no additional attributes that need to be copied over except for self.__dict__